elpy.el

```diff
@@ -3292,13 +3293,161 @@ Returns a possible multi-line docstring."
+    (save-excursion
+      (goto-char (xref-elpy-location-pos l))
+      (point-marker))))
+
+(cl-defmethod xref-item-location ((l xref-elpy-location))
+  (with-current-buffer (find-file-noselect (xref-elpy-location-file l))
+    (save-excursion
+      (goto-char (xref-elpy-location-pos l))
+      (point-marker))))
+
+(cl-defmethod xref-location-group ((l xref-elpy-location))
+  (xref-elpy-location-file l))
+
+;; Get identifier
+(cl-defmethod xref-backend-identifier-at-point ((_backend (eql elpy)))
+  (elpy-xref--identifier-at-point))
+
+(defun elpy-xref--identifier-at-point ()
+  "Return identifier at point.
+Currently use the position as identifier"
+  (point))
+
+;; Jump to definition
+(cl-defmethod xref-backend-definitions ((_backend (eql elpy)) symbol)
+  (elpy-xref--definitions symbol))
+
+(defun elpy-xref--definitions (symbol)
+  "Return the symbol definition position"
+  (save-excursion
+    (goto-char symbol)
+    (let* ((location (elpy-rpc-get-definition)))
+      (if (not location)
+          ;; Just throw an error, but should return the list of all possible identifiers
+          (error "No definition found")
+        (let* ((file (expand-file-name (car location)))
+               (pos (+ 1 (car (cdr location))))
+               (line (with-current-buffer (find-file-noselect file)
+                       (goto-char (+ pos 1))
+                       (buffer-substring (line-beginning-position) (line-end-position))))
+               (loc (xref-make-elpy-location file pos line))
+               (summary (format elpy-xref--format-references pos file)))
+          (list (xref-make summary loc)))))))
+
+;; Get usages
+(cl-defmethod xref-backend-references ((_backend (eql elpy)) symbol)
+  (elpy-xref--references symbol))
+
+(defvar elpy-xref--format-references
+  (let ((str "%s:\t%s"))
+    (put-text-property 0 4 'face 'font-lock-comment-face str)
+    str)
+  "Formatting for references")
+
+(defun elpy-xref--references (symbol)
+  "Return the symbol references"
+  (save-excursion
+    (goto-char (string-to-int symbol))
+    (let* ((references (elpy-rpc-get-usages)))
+      (cl-loop
+       for ref in references
+       for name = (alist-get 'name ref)
+       for file = (alist-get 'filename ref)
+       for pos = (alist-get 'offset ref)
+       for linenumber = (with-current-buffer (find-file-noselect file)
+                          (line-number-at-pos pos))
+       for line = (with-current-buffer (find-file-noselect file)
+                    (save-excursion
+                      (goto-char (+ pos 1))
+                      (buffer-substring (line-beginning-position) (line-end-position))))
+       for summary = (format elpy-xref--format-references linenumber line)
+       for loc = (xref-make-elpy-location file (+ pos 1) line)
+       collect (xref-make summary loc)))))
+
+(defun xref-find-references ()
+  "Overwrite the current `xref-find-references` to avoid completion asking"
```

Comment:
Isn't this kind of rude, unconditionally redefining a function in another library? This affects non-Elpy uses of xref, too.